### ISSUE TITLE
Cloud Controller Manager Unable to Split Provider ID for VM in Availability Set in Azure

### ISSUE THREAD

**Issue by marc-sensenich** [kubernetes/kubernetes#70126]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
When creating a Kubernetes 1.12.1 cluster in Azure with Availability Set backed masters and VMSS backed agents, the cloud-controller-manager is unable to parse the ProviderID returned by the Azure SDK.

```sh
# This line should have a `azure://` in front of it based on the providerIDRE for 1.12.1
I1023 11:45:27.073599       1 azure_vmss.go:196] Can not extract scale set name from providerID (/subscriptions/***/resourceGroups/***/providers/Microsoft.Compute/virtualMachines/master-0), assuming it is mananaged by availability set: not a vmss instance
E1023 11:45:27.073624       1 node_controller.go:276] Error checking if node master-0 exists: error splitting providerID
I1023 11:45:27.073683       1 azure_instances.go:122] NodeAddressesByProviderID: omitting unmanaged node "azure:///subscriptions/***/resourceGroups/***/providers/Microsoft.Compute/virtualMachines/master-0"
E1023 11:45:27.073707       1 node_controller.go:419] failed to find kubelet node IP from cloud provider

```

**What you expected to happen**:

The providerID should be split and the cloud-controller-manager should be able to find a kubelet node IP from the cloud provider

**How to reproduce it (as minimally and precisely as possible)**:
- Stand up a master node in Azure within an availability set with the kubelet configured with `--cloud-provider=external` 
- Run the `cloud-controller-manager` configured for Azure

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:46:06Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:36:14Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: Azure
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`): `4.15.0-1023-azure #24~16.04.1-Ubuntu SMP Wed Aug 29 12:54:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
- Install tools:
- Others:

**Comment by marc-sensenich** [comment]
When switching over to vmType `standard` the following was encountered; `I1023 16:31:33.198066       1 azure_instances.go:122] NodeAddressesByProviderID: omitting unmanaged node "azure:///subscriptions/***/resourceGroups/***/providers/Microsoft.Compute/virtualMachines/master-0`

**Comment by marc-sensenich** [comment]
The fix in #70135 resolved the issue that was seen in this issue, closing
